client/data/home/use-home-layout-query-params.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { useSelector } from 'react-redux';
+import config from '@automattic/calypso-config';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentQueryArguments } from 'calypso/state/selectors/get-current-query-arguments';
+
+interface HomeLayoutQueryParams {
+	dev?: true;
+	view?: string;
+}
+
+export function useHomeLayoutQueryParams(): HomeLayoutQueryParams {
+	const { dev, view } = useSelector( getCurrentQueryArguments ) as any;
+
+	return {
+		dev: config.isEnabled( 'home/layout-dev' ) || dev === 'true' || undefined,
```

Comment:
I know this was like that before, but I'd like to point out that it can be confusing to have it enabled by either a feature flag or a query argument. Usually, when I enable a feature flag, I expect that the related functionality will be triggered. Likewise, if I disable it, I expect the related functionality will not be triggered - but in this instance, the opposite will happen. Maybe that's just fine, but I just wanted to point out it feels odd from an external perspective 😉

Reply:
Now that you mention it I would expect the explicit query param to take precedence over the config i.e. I can use `?dev=false` to override the config. The short answer is that this situation hasn't come up.

[We thought about removing the dev flag](https://github.com/Automattic/wp-calypso/pull/53207#discussion_r654198636). I want to keep it around because I can see it being useful for future backend changes, but given it's not used for anything ATM we can change the logic to whatever makes sense.

I'm going to make the query param take precedence.